wordpress-mobile/WordPress-Android
# Jetpack Section: Update Activity Log Details UI and Add Backup Download Button/Flow

Fixes #14095

This PR does the below:
- It updates the `Activity Log Details` screen UI based on design (see [this comment](https://github.com/wordpress-mobile/WordPress-Android/issues/14095#issuecomment-780685652) for more details).
- It add the `Download Backup` button and connects the `Activity Log Details` screen with the backup download flow.
- It replaces the `Rewind Status Service` with a more simplified solution for making sure that the restore and download backup buttons are shown/hidden depending on the rewindable and in progress status of the activity log items.

More importantly, this PR makes sure that when a user presses the `DOWNLOAD BACKUP` button within the `Activity Log Detail` screen, then they get redirected to the `Backup Download` screen. Also, when a backup download gets triggered from within this flow, this PR makes sure that the user get redirected back to the `Activity Log` screen, skipping the `Activity Log Detail` screen and starts seeing the backup download progress row within that screen.

Light Theme | Dark Theme | Layout Bounds
-------|------|------
<img width="250" height="500" alt="before" src="https://user-images.githubusercontent.com/9729923/108517573-41f0db80-72d0-11eb-82ec-5cae4d54bcde.png"> | <img width="250" height="500" alt="after" src="https://user-images.githubusercontent.com/9729923/108517597-4b7a4380-72d0-11eb-8316-0ee557bb5a94.png"> | <img width="250" height="500" alt="after" src="https://user-images.githubusercontent.com/9729923/108517630-559c4200-72d0-11eb-827a-0c1c769d2df9.png">

To test:
* Open app and go to `My Site` tab.
* Launch `Activity Log` or `Backup` screen.
* Find a rewindable item from the list and click on it. You will now enter the `Activity Log Detail` screen.
* On that screen, make sure that both, the `RESTORE` and `DOWNLOAD BACKUP` buttons are visible.
* Make sure the UI matches the designs (see [this comment](https://github.com/wordpress-mobile/WordPress-Android/issues/14095#issuecomment-780685652) for more details).
* Go back to the previous list screen.
* Find a non-rewindable item from the list and click on it. You will now enter the `Activity Log Detail` screen.
* On that screen, make sure that both, the `RESTORE` and `DOWNLOAD BACKUP` buttons are not visible.
* Go back to the previous list screen.
* Trigger the `Restore` or `Backup Download` flow from a rewindable item from the list. Wait for the progress item to appear and the menu item on all the list items to disappear.
* While the `Restore` or `Backup Download` process is in progress, find a rewindable item from the list and click on it. You will now enter the `Activity Log Detail` screen.
* On that screen, make sure that both, the `RESTORE` and `DOWNLOAD BACKUP` buttons are *NOT* visible.
* Go back to the previous list screen.
* While the `Restore` or `Backup Download` process is in progress, find a non-rewindable item from the list and click on it. You will now enter the `Activity Log Detail` screen.
* On that screen, make sure that both, the `RESTORE` and `DOWNLOAD BACKUP` buttons are not visible.
* Wait for the `Restore` or `Backup Download` process to finish.
* Find a rewindable item from the list and click on it. You will now enter the `Activity Log Detail` screen.
* On that screen, click the `RESTORE` button to launch the `Restore` screen,
* Click the `Restore site` button. On the confirmation screen, click on the `Confirm restore site` button.
* Navigate back manually (by clicking the `X` toolbar button or pressing/swiping back).
* You should automatically be navigated back to the `Activity Log` screen, skipping the `Activity Log Detail` screen and a restore status will be triggered,
* The restore progress item should immediately appears. Make sure the restore timestamp is available on this progress item.
* Try doing the same steps again, but this time when within the `Restore` screen and flows, **DO NOT** trigger the restore, go back.
* Make sure that going back will navigate to the `Activity Log Detail` screen.
* Again, find a rewindable item from the list and click on it. You will now enter the `Activity Log Detail` screen.
* On that screen, click the `DOWNLOAD BACKUP` button to launch the `Backup Download` screen,
* Click the `Create downloadable file` button.
* Navigate back manually (by clicking the `X` toolbar button or pressing/swiping back).
* You should automatically be navigated back to the `Activity Log` screen, skipping the `Activity Log Detail` screen and a backup download status will be triggered,
* The backup download progress item should immediately appears. Make sure the backup download timestamp is available on this progress item.
* Try doing the same steps again, but this time when within the `Backup Download` screen and flows, **DO NOT** trigger the backup download, go back.
* Make sure that going back will navigate to the `Activity Log Detail` screen.

PR submission checklist:
